### PR TITLE
[New Rule] Azure Event Hub Deletion

### LINKED ISSUE #128: [New Rule] Azure Event Hub Deletion
## Description
Identifies an Event Hub deletion. In Azure, an Event Hub is an event processing service that ingests and processes large volumes of events and data. An adversary may delete an Event Hub in an attempt to evade defenses.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`azure`

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about
https://azure.microsoft.com/en-in/services/event-hubs/
https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features

## Syntax
```
event.module:azure and event.dataset:azure.activitylogs and event.category:Administrative and azure.activitylogs.operation_name:MICROSOFT.EVENTHUB/NAMESPACES/EVENTHUBS/DELETE and event.outcome:Success
```

## Example Data
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issue(s)
resolves #128 

## Description

Identifies an Event Hub deletion in Azure. An Event Hub is an event processing service that ingests and processes large volumes of events and data. An adversary may delete an Event Hub in an attempt to evade defenses.

## Checklists
Use ~~strikethroughs~~ to remove items which are not applicable to this issue.

### For submitter
- [x] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
  -- Query field process.name.text doesn't currently exist in endgame-* eventing data, but does exist for winlogbeats, and is case insensitive.
- [x] Verify all query fields co-exist within a single ECS version and annotated the minimum version
  -- 1.4.0
- [ ] Verified the query detects the intended event(s)
  - detonate
  - search in _discover_
  - trigger as a custom signal
- [x] Create rule using `create-rule`
- [x] Verify and convert to standard linting (`toml-lint -f <rule-file>`)
- [x] Run tests using `pytest -x -v unit_tests` or `make test`
- [ ] Internal search to determine noise and FP

### For reviewers
- [ ] Verify existing rule for activity doesn't exist as a siem or endpoint rule (unless intentionally specified in issue)
- [ ] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
- [ ] Verify all query fields co-exist within a single ECS version and annotated the minimum version
- [ ] Internal search to determine noise and FP
- [ ] Verify metadata accuracy and spelling

### CODE REVIEW

File: rules/azure/defense_evasion_event_hub_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2020/08/18"
+ecs_version = ["1.5.0"]
```
Comment (threat-punter) [inline rules/azure/defense_evasion_event_hub_deletion.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

File: rules/azure/defense_evasion_event_hub_deletion.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+maturity = "production"
+updated_date = "2020/08/18"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies an Event Hub deletion in Azure. An Event Hub is an event processing service that ingests and processes large
+volumes of events and data. An adversary may delete an Event Hub in an attempt to evade defenses.
```
Comment (threat-punter) [inline rules/azure/defense_evasion_event_hub_deletion.toml:11]: Minor suggestion. Maybe detection would be more accurate?

```suggestion
volumes of events and data. An adversary may delete an Event Hub in an attempt to evade detection.
```

**Review by threat-punter — APPROVED** [review]
🤗

**Review by brokensound77 — APPROVED** [review]
suggestions and LGTM

**Pull request merged**